zone/bot_command.cpp

```diff
@@ -2119,42 +2089,146 @@ bool helper_spell_check_fail(STBaseEntry* local_entry)
+		bot_owner->Message(Chat::Yellow, "%s", required_bots_map[spell_type].c_str());
 		return true;
 	}
 
 	return false;
 }
 
+void SendSpellTypeWindow(Client* c, const Seperator* sep) {
+	std::string arg0 = sep->arg[0];
+	std::string arg1 = sep->arg[1];
+
+	uint8 minCount = 0;
+	uint8 maxCount = 0;
+	bool clientOnly = false;
+
+	if (BotSpellTypes::END <= 19) {
+		minCount = BotSpellTypes::START;
+		maxCount = BotSpellTypes::END;
+	}
+	else if (!arg1.compare("0-19")) {
+		minCount = BotSpellTypes::START;
+		maxCount = 19;
+	}
+	else if (!arg1.compare("20-39")) {
+		minCount = std::min(static_cast<uint8_t>(20), static_cast<uint8_t>(BotSpellTypes::END));
+		maxCount = std::min(static_cast<uint8_t>(39), static_cast<uint8_t>(BotSpellTypes::END));
+	}
+	else if (!arg1.compare("40+")) {
+		minCount = std::min(static_cast<uint8_t>(40), static_cast<uint8_t>(BotSpellTypes::END));
+		maxCount = BotSpellTypes::END;
+	}
+	else if (!arg1.compare("commanded")) {
+		minCount = BotSpellTypes::COMMANDED_START;
+		maxCount = BotSpellTypes::COMMANDED_END;
+	}
+	else if (!arg1.compare("client")) {
+		minCount = BotSpellTypes::START;
+		maxCount = BotSpellTypes::END;
+		clientOnly = true;
+	}
+	else {
+		c->Message(Chat::Yellow, "You must choose a valid range option");
+
+		return;
+	}
+
+	const std::string& indian_red = "indian_red";
+	const std::string& gold = "gold";
+	const std::string& slate_blue = "slate_blue";
+	const std::string& forest_green = "forest_green";
+	const std::string& goldenrod = "goldenrod";
+
+	std::string fillerLine = "-----------";
+	std::string spellTypeField = "Spell Type";
+	std::string idField = "ID";
+	std::string shortnameField = "Short Name";
+
+	std::string popup_text = DialogueWindow::TableRow(
+		DialogueWindow::TableCell(
+			fmt::format(
+				"{}",
+				DialogueWindow::ColorMessage(goldenrod, spellTypeField)
+			)
+		) +
+		DialogueWindow::TableCell(
+			fmt::format(
+				"{}",
+				(!arg0.compare("^spelltypeids") ? DialogueWindow::ColorMessage(goldenrod, idField) : DialogueWindow::ColorMessage(goldenrod, shortnameField))
+			)
+		)
+	);
+
+	popup_text += DialogueWindow::TableRow(
+		DialogueWindow::TableCell(
+			fmt::format(
+				"{}",
+				DialogueWindow::ColorMessage(gold, fillerLine)
+			)
+		) +
+		DialogueWindow::TableCell(
+			fmt::format(
+				"{}",
+				DialogueWindow::ColorMessage(gold, fillerLine)
+			)
+		)
+	);
+
+	for (int i = minCount; i <= maxCount; ++i) {
+		if (clientOnly && !IsClientBotSpellType(i)) {
+			continue;
+		}
+
+		popup_text += DialogueWindow::TableRow(
+			DialogueWindow::TableCell(
+				fmt::format(
+					"{}",
+					DialogueWindow::ColorMessage(forest_green, c->GetSpellTypeNameByID(i))
+				)
+			) +
+			DialogueWindow::TableCell(
+				fmt::format(
+					"{}",
+					(!arg0.compare("^spelltypeids") ? DialogueWindow::ColorMessage(slate_blue, std::to_string(i)) : DialogueWindow::ColorMessage(slate_blue, c->GetSpellTypeShortNameByID(i)))
+				)
+			)
+		);
+	}
```

Comment:
Don't need to use `fmt::format` when the returned value is a string. Can just do `DialogueWindow::TableCell(DialogueWindow::ColorMessage("blah"));`.